src/table.jl

```diff
@@ -31,7 +31,7 @@ mutable struct DNextTable{T,K} <: AbstractIndexedTable
     # primary key columns
     pkey::Vector{Int}
     # extent of values in the pkeys
-    domains::Vector{IndexSpace{K}}
+    domains::Vector{IndexSpace}
```

Comment:
This would make `domains` abstractly typed here and lead to problems with inferring functions on `DNextTable`, right? Is that intended?

Reply:
Yes, this was intentional. Discussed with @shashi

Reply:
We still keep the `K`, just not require the IndexSpace to have tuple of the exact same type as K.